File: src/main/java/com/github/puregero/warpplugin/DelWarpCommand.java
package com.github.puregero.warpplugin;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabCompleter;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;
import java.util.Objects;

public class DelWarpCommand implements CommandExecutor, TabCompleter {
    private final WarpPluginPlugin plugin;

    public DelWarpCommand(WarpPluginPlugin plugin) {
        this.plugin = plugin;
        Objects.requireNonNull(plugin.getCommand("delwarp")).setExecutor(this);
        Objects.requireNonNull(plugin.getCommand("delwarp")).setTabCompleter(this);
    }

    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (args.length == 0) {
            sender.sendMessage(Component.text("Usage: /%s <warp>".formatted(label)).color(NamedTextColor.RED));
            return false;
        }

        String warpName = args[0];

        synchronized (this.plugin) {
            this.plugin.getConfig().set("warps." + warpName, null);
        }

        this.plugin.scheduleConfigSave();

        sender.sendMessage(Component.text("Warp deleted: " + warpName).color(NamedTextColor.GREEN));

        return true;
    }

    @Override
    public @Nullable List<String> onTabComplete(@NotNull CommandSender commandSender, @NotNull Command command, @NotNull String s, @NotNull String[] args) {
        if (args.length == 1) {
            return this.plugin.getWarpNames().stream().filter(warpName -> warpName.toLowerCase().startsWith(args[0].toLowerCase())).toList();
        }
        return List.of();
    }
}

File: src/main/java/com/jjkay03/foliawarp/commands/DelWarpCommand.kt
package com.jjkay03.foliawarp.commands

import com.jjkay03.foliawarp.FoliaWarp
import com.jjkay03.foliawarp.Utils
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.command.TabCompleter

class DelWarpCommand : CommandExecutor, TabCompleter {

    init {
        FoliaWarp.INSTANCE.getCommand("delwarp")?.setExecutor(this)
        FoliaWarp.INSTANCE.getCommand("delwarp")?.tabCompleter = this
    }

    // COMMAND
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
        // End if invalid arg
        if (args.isEmpty()) { sender.sendMessage("§cUsage: /$label <warp>"); return false }

        val warpName = args[0]

        synchronized(FoliaWarp.INSTANCE) {
            FoliaWarp.INSTANCE.config.set("warps.$warpName", null)
        }

        Utils.scheduleConfigSave()

        // Feedback
        sender.sendMessage("§3\uD83C\uDF00 §bDeleted warp §3$warpName")

        return true
    }

    // TAB COMPLETION
    override fun onTabComplete(sender: CommandSender, command: Command, label: String, args: Array<out String>): List<String>? {
        if (args.size == 1) {
            return Utils.getWarpNames().filter { it.lowercase().startsWith(args[0].lowercase()) }
        }
        return emptyList()
    }

}
